### ISSUE TITLE
Nonce with equals issue

### ISSUE THREAD

**Issue by naumansiddiqui4** [ibnesayeed/webserver-tester#39]
When I create nonce from server which contains a = in the nonce. It fails my test case parse_equal_sign_delimited_keys_values in a4 test suite.

I checked web-tester code in httptester.py and function parse_equal_sign_delimited_keys_values, code parses for key and values by splitting on =. This, I feel could be issue for my test case failing.

If my nonce does not have an equals sign, then my test case never fails.

**Comment by ibnesayeed** [comment]
Yes, that indeed is a bug. I will fix the `WWW-Authenticate` header parser.

**Comment by ibnesayeed** [comment]
By the way, do you have a specific situation where you want to have `=` sign in the values of attributes of `WWW-Authenticate` header?

**Comment by naumansiddiqui4** [comment]
I do not have any specific situation. When I run the test case multiple
times, it fails once because of the equals sign for nonce value.

Thanks & Regards,
Mohammed Nauman Siddique

Contact  Number: +91- 8826206356

**Comment by ibnesayeed** [comment]
While I understand that quoted strings may contain `=` sign, the recommended way of generating nonce in this class is a Hex Digest which should not contain it. So, far the sake of simplicity, I would defer this fix for a later day, unless necessary.
